src/util/SystemCommand.ts

```diff
@@ -184,31 +187,55 @@ export class SystemCommand {
                     let errorDetail = '';
+                    const pluginCmd = this.args[0];
+                    const output = this.outputChannel;
+
+                    // NOTE: Since we are using callback promises we need to use `bind` method to ensure that 
+                    // we are using the original `this` context (via `self`) that called executeOuputChannel
+                    const rerunCmd = this.executeWithOutputChannel.bind(self);
+
                     switch (condition) {
                         case CommandDetection.ERR_COMMAND_NOT_FOUND:
                             errorDetail = `Unable to locate '${this.command}' command.`;
                             break;
                         case CommandDetection.ERR_PLUGIN_NOT_FOUND:
-                            errorDetail = `Unable to locate '${this.command}' '${this.args[0]}' plugin.`;
+                            errorDetail = `Unable to locate '${this.command}' '${this.args[0]}' plugin. Would you like to install '${this.args[0]}' and rerun the previous command?`;
+                            // ask user if we should install the missing plugin and rerun the previous command
+                            window.showQuickPick([CONFIRM_YES, CONFIRM_NO], { title: errorDetail })
+                                .then((res:string) => {
+                                    if (res === CONFIRM_YES) {
+                                        installPlugin(pluginCmd, output)
+                                            .then((status:number) => {
+                                                if (status === 0) {
+                                                    resolve(rerunCmd());
+                                                }
+                                                resolve(status);
+                                            });
+                                    }
+                                });
+
                             break;
                     }
-                    window.showErrorMessage(errorDetail);
 
-                    errorDetail += '\nFor additional detail, please see https://cloud.ibm.com/docs/cli';
-                    this.output(errorDetail);
+                    // Only print addition error messages for non plugin errors
+                    if (condition != CommandDetection.ERR_PLUGIN_NOT_FOUND) {
+                        errorDetail += '\nFor additional detail, please see https://cloud.ibm.com/docs/cli';
+                        window.showErrorMessage(errorDetail);
+                        this.output(errorDetail);
+
+                        // add this at the end of the promise chain, so it gets called last, for certain.
+                        // this allows us to fulfill other steps in the promise chain, and clean everything up last
+                        promise.finally(function() {
+                            self.destroy();
+                        });
```

Comment:
**Callout**: Moved destroy into conditional block so that the attributes are not undefined when trying to rerun previous failed command. Running previous command only occurs after install a missing plugin.